dashboard/src/components/DeploymentFormBody/DifferentialTab.test.tsx

```diff
@@ -0,0 +1,91 @@
+import { CdsIcon } from "@clr/react/icon";
+import { mount } from "enzyme";
+import * as React from "react";
+import { act } from "react-dom/test-utils";
+import DifferentialTab from "./DifferentialTab";
+
+describe("when installing", () => {
+  it("should show the changes icon if the values change", () => {
+    const wrapper = mount(
+      <DifferentialTab
+        deploymentEvent="install"
+        deployedValues=""
+        defaultValues="foo"
+        appValues="bar"
+      />,
+    );
+    expect(wrapper.find(CdsIcon).prop("hidden")).toBe(false);
+  });
+
+  it("should hide the changes icon if the values are the same", () => {
+    const wrapper = mount(
+      <DifferentialTab
+        deploymentEvent="install"
+        deployedValues=""
+        defaultValues="foo"
+        appValues="foo"
+      />,
+    );
+    expect(wrapper.find(CdsIcon).prop("hidden")).toBe(true);
+  });
+
+  it("clicking the tab removes the icon", () => {
+    const wrapper = mount(
+      <DifferentialTab
+        deploymentEvent="install"
+        deployedValues=""
+        defaultValues="foo"
+        appValues="bar"
+      />,
+    );
+    expect(wrapper.find(CdsIcon).prop("hidden")).toBe(false);
+    act(() => {
+      wrapper.find("div").simulate("click");
+    });
+    wrapper.update();
+    expect(wrapper.find(CdsIcon).prop("hidden")).toBe(true);
+  });
+
+  it("settind default values removes the icon", () => {
+    const wrapper = mount(
+      <DifferentialTab
+        deploymentEvent="install"
+        deployedValues=""
+        defaultValues="foo"
+        appValues="bar"
+      />,
+    );
+    expect(wrapper.find(CdsIcon).prop("hidden")).toBe(false);
+    act(() => {
+      wrapper.setProps({ appValues: "foo" });
+    });
+    wrapper.update();
+    expect(wrapper.find(CdsIcon).prop("hidden")).toBe(true);
+  });
+});
+
+describe("when upgrading", () => {
+  it("should show the changes icon if the values change", () => {
+    const wrapper = mount(
+      <DifferentialTab
+        deploymentEvent="upgrade"
+        deployedValues="foo"
+        defaultValues="foo"
+        appValues="bar"
+      />,
+    );
+    expect(wrapper.find(CdsIcon).prop("hidden")).toBe(false);
+  });
+
+  it("should hide the changes icon if the values are the same", () => {
+    const wrapper = mount(
+      <DifferentialTab
+        deploymentEvent="upgrade"
+        deployedValues="foo"
+        defaultValues="foo"
+        appValues="foo"
+      />,
+    );
+    expect(wrapper.find(CdsIcon).prop("hidden")).toBe(true);
+  });
```

Comment:
Is there something missing here for hitting the default values (or clearing changes) should have on the icon, when upgrading?

Reply:
yes, thats https://github.com/kubeapps/kubeapps/pull/2032/commits/a817abc14f42eb9f92d499ada72abc52a6c48b5b. When clicking the "restore" button, `oldValues` is equal to `appValues` removing the icon.